pkg/kubernetes/scripts/virtual-machines/components/createVmDialog.jsx

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from 'cockpit';
+import React from "react";
+
+const _ = cockpit.gettext;
+
+import './createVmDialog.less';
+
+
+class CreateVmDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            resource: '',
+            resourceError: null,
+        };
+
+        this.onResourceChanged = this.onResourceChanged.bind(this);
+        this.getResult = this.getResult.bind(this);
+        this.validate = this.validate.bind(this);
+    }
+
+    onResourceChanged(e) {
+        if (e && e.target && typeof(e.target.value) !== 'undefined') {
+            this.setState({ resource: e.target.value });
+        }
+    }
+
+    validate() {
+        let validation = true;
+
+        let resourceError = null;
+
+        try {
+            if (!this.state.resource) {
+                resourceError = _("VM definition is required.");
+            } else {
+                let res = JSON.parse(this.state.resource);
+                if (res instanceof Array) {
+                    resourceError = _("VM definition should be an object. Not an array.");
```

Comment:
"should" → "must", and please check for other wrong types too (e. g. `number` and `string`). Leave the "not an ..." part out, as this would otherwise need to be translated for every wrong type, and it's  not very interesting.

Even better would be to do a positive type test. Simply testing for `instanceof Object` does not work of course, but how about `"kind" in res`?

Reply:
in latest commits it is parsing + checking for an object and asking for valid JSON in return

Reply:
> but how about "kind" in res

the semantics is left for the backend (kube-client) to check: this will result in: 
`the server could not find the requested resource`